src/Framework/IEventSource5.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable disable
+
+namespace Microsoft.Build.Framework
+{
+    /// <summary>
+    /// This interface defines the events raised by the build engine.
+    /// Loggers use this interface to subscribe to the events they
+    /// are interested in receiving.
+    /// </summary>
+    public interface IEventSource5 : IEventSource4
```

Comment:
So this is undoubtedly the _right_ thing to do for this. However there is an easier way that has less downstream impact: a new BuildMessageEventArgs subclass that we just log and treat specially in the TerminalLogger. Did you consider that?

Reply:
I removed the v5.